File: src/dip/python/messageSender.py
# Pas d'interface formelle
class EmailSender:
    def send(self, message):
        print(f"Email envoyé : {message}")

class NotificationService:
    def __init__(self, sender):
        self.sender = sender

    def notify(self, message):
        self.sender.send(message)

File: src/dip/dart/messageSender.dart
abstract class MessageSender {
  void send(String message);
}

class EmailSender implements MessageSender {
  void send(String message) {
    print("Email envoyé : $message");
  }
}

class NotificationService {
  final MessageSender sender;

  NotificationService(this.sender);

  void notify(String message) {
    sender.send(message);
  }
}
